### PR TITLE
[Fix] Added documentation link for the dev tools' help menu 

### LINKED ISSUE #4645: [BUG] Update Dev Tools Documentation link under help in console
**Describe the bug**

Update documentation link to Dev Tools documentation. Link currently goes to general OpenSearch Dashboards documentation. Link should go to Dev Tools documentation at https://opensearch.org/docs/latest/dashboards/dev-tools/index-dev/

**To Reproduce**
1. Go to 'OpenSearch Dashboards' main menu.
2. Select on 'Dev Tools'
3. In the Console, select the Help icon (question mark within a circle), then select Documentation. 

**Expected behavior**
Reader should be directed to the Dev Tools documentation main page.

**Dashboards Version**
2.9

**Screenshots**
Screenshot 2023-07-31 at 3 22 35 PM

### PR DESCRIPTION
**Pull request by Willie-The-Lord**
### Description

Added Help Menu extension (dev tool documentation link - https://opensearch.org/docs/latest/dashboards/dev-tools/index-dev/) to dev tool plugin's help menu.

### Issues Resolved

Added dev tool's documentation link to the dev tools' help menu 
closes #4645 

## Screenshot

The documentation link appears correctly in my local environment.

- Before
Screenshot 2023-10-11 at 12 17 02 PM
- After
Screenshot 2023-10-11 at 12 18 10 PM

### Check List

- [ ] All tests pass
  - [X] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ashwin-pc** [comment]
@Willie-The-Lord Thanks for jumping on this really quickly. Looks like the original issue was misleading. The link it asked you to update is a global link and not specific to Dev Tools. To add a dev tools specific link you need to add a new one instead of modifying one. Here is an example of the Visualize app and how it adds such a link: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/visualize/public/application/utils/utils.ts#L37

Also be sure to add a changelog entry for your PR.

**Comment by ananzh** [comment]
Also need a CHANGELOG. Here is some guidance https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5170#issuecomment-1743705149

**Comment by Willie-The-Lord** [comment]
Hi @ashwin-pc I've added the new documentation link to the dev tool help menu. I've looked up other plugins (Visualize & Dashboards) and found out the function `addHelpMenuToAppChrome` is placed in an [util folder](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/visualize/public/application/utils). Hence I created one in the dev tool plugin too. Not really sure whether it is the best practice. Thanks!

**Comment by Willie-The-Lord** [comment]
I found out that I've signed off other commits that doesn't belongs to me. How can I reverse that?

**Comment by Willie-The-Lord** [comment]
Hi @ananzh @ashwin-pc, I've just rebased the commit and updated the PR. Thanks!

File: src/plugins/dev_tools/public/utils/util.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
```
Comment (ananzh) [inline src/plugins/dev_tools/public/utils/util.ts:1]: For new created file, you could use license 
```
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -67,6 +67,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [BUG][Data Explorer][Discover] Allow filter and query persist when refresh page or paste url to a new tab ([#5206](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5206))
 - [Data Explorer] Remove the `X` icon in data source selection field ([#5238](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5238))
 - [BUG][Fuctional Test] Make setDefaultAbsoluteRange more robust and update doc views tests ([#5242](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5242))
+- [BUG][Dev Tool] Added dev tool documentation link to dev tool's help menu [#5166](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5166)
```
Comment (ananzh) [inline CHANGELOG.md:70]: NIT: Maybe changing `Added` to `Add` is more consistent with other CHANGELOG items.

**Review by ananzh — COMMENTED** [review]
Cool. Just need some nit changes. Could you add two screenshots before & after your changes?

**Comment by Willie-The-Lord** [comment]
Hi @ananzh, I've updated the code. Thanks!

**Review by ananzh — DISMISSED** [review]

**Comment by ashwin-pc** [comment]
@Willie-The-Lord looks like a few tests are failing because of snapshots being out of date. Can you run yarn test:jest -u locally and update the snapshots too? We have a few tests that unfortunately add to the snapshot data all the available doc links. Adding the new link means that now we have to update them as well.

**Comment by ashwin-pc** [comment]
@Willie-The-Lord you latest commit did not have a signoff thats tripping the DCO checker. Can you follow the instruction in the DCO workflow to fix it?

File: src/dev/build/tasks/os_packages/create_os_package_tasks.test.ts [hunk 3]
```diff
@@ -158,6 +158,22 @@ it('runs task with the expected values for RPM x64', async () => {
             "linux",
           ],
         ],
+        Array [
```
Comment (ruanyl) [inline src/dev/build/tasks/os_packages/create_os_package_tasks.test.ts:161]: I may not modify tests which is unrelated to the PR changes. Just curious why changing these tests?

Comment (Willie-The-Lord) [reply src/dev/build/tasks/os_packages/create_os_package_tasks.test.ts:161]: Sorry about that. I'm just testing something and making the commit history kind of messy. I will change it later today.

**Comment by Willie-The-Lord** [comment]
Hi @ashwin-pc, I ran `yarn test:jest -u` locally, and it modified several files in the process. Now, I'm unsure which files I need to push to the remote repository. Any guidance on this would be greatly appreciated!

**Comment by joshuarrrr** [comment]
> Hi @ashwin-pc, I ran `yarn test:jest -u` locally, and it modified several files in the process. Now, I'm unsure which files I need to push to the remote repository. Any guidance on this would be greatly appreciated!

You probably want to push all of them. But you can also check the prior workflow to see which previous failures needed to be fixed.

**Comment by Willie-The-Lord** [comment]
Hi @joshuarrrr @ashwin-pc , I've pushed the new snapshots.

**Review by ashwin-pc — APPROVED** [review]
Nice! LGTM :)

**Comment by Willie-The-Lord** [comment]
Thanks @ashwin-pc ! It seems like all the test cases passed.

**Review by joshuarrrr — APPROVED** [review]

**Pull request merged**